lighthouse-core/gather/driver.js

```diff
@@ -275,6 +268,121 @@ class Driver {
     this._nextProtocolTimeout = timeout;
   }
 
+  /**
+   * @param {LH.Protocol.RawEventMessage} event
+   */
+  _handleProtocolEvent(event) {
+    this._devtoolsLog.record(event);
+    if (this._networkStatusMonitor) {
+      this._networkStatusMonitor.dispatch(event);
+    }
+
+    // @ts-ignore TODO(bckenny): tsc can't type event.params correctly yet,
+    // typing as property of union instead of narrowing from union of
+    // properties. See https://github.com/Microsoft/TypeScript/pull/22348.
+    this._eventEmitter.emit(event.method, event.params);
+  }
+
+  /**
+   * @param {Error} error
+   */
+  _handleEventError(error) {
+    log.error('Driver', 'Unhandled event error', error.message);
+  }
+
+  /**
+   * @param {LH.Crdp.Target.ReceivedMessageFromTargetEvent} event
+   * @param {string[]} parentSessionIds The list of session ids of the parents from youngest to oldest.
+   */
+  async _handleReceivedMessageFromTarget(event, parentSessionIds) {
+    const {sessionId, message} = event;
```

Comment:
need to use this `sessionId` below (or don't bind it here) for linting

Reply:
oh whoops yeah thanks!